modules/salt/pages/salt-rate-limiting.adoc

```diff
@@ -0,0 +1,62 @@
+[[salt.rate.limiting]]
+= Rate Limiting
+
+Salt is able to run commands in parallel on a large number of minions.
+This can potentially create large amounts of load on your infrastructure.
+You can use these rate-limiting parameters to control the load in your environment.
+
+These parameters are all configured in the [filename]``/etc/rhn/rhn.conf`` configuration file.
+
+[NOTE]
+====
+Salt commands that are executed from the command line are not subject to these parameters.
+====
+
+
+
+== Batch Size
+
+This parameter sets the maximum number of clients that can execute a single action at the same time.
+
+Choosing a lower number will reduce system load and parallelism, but might reduce overall performance for processing actions.
+
+Adjust the [systemitem]``java.salt_batch_size`` parameter.
+Defaults to 100.
+
+
+
+== Batch Delay
+
+This parameter sets the amount of time, in seconds, to wait after an action is processed before beginning to process the action on the next minion.
+To begin, actions are processed on the number of minions determined in the batch size parameter.
+As each minion completes, the action begins on a new minion after the specified delay period.
+
+Increasing the delay increases the chance that multiple minions will have completed before the next action is issued, resulting in fewer overall commands, and reducing load.
+
+Adjust the [systemitem]``java.salt_batch_delay`` parameter.
+Defaults to 1.0 seconds.
+
+
+
+== Ping Timeout
+
+During Salt batch calls, Salt clients will perform a presence ping.
+This parameter changes the timeout setting for the presence ping, in seconds.
+
+Adjust the [systemitem]``java.salt_presence_ping_timeout`` parameter.
+Defaults to 4 seconds.
+
+
+
+== Ping Gather Job Timeout
+
+During Salt batch calls, a job runs to handles the presence ping from the clients.
+This parameter changes the timeout setting for gathering the presence ping, in seconds.
+
+If a minion does not reply to a ping within the allocated time, it will be marked as [systemitem]``not available``, and will be excluded from the command.
+The {webui} will show a [systemitem]``minion is down`` message in this case.
+
+For more information on minion timeouts, see xref:scale-minions.adoc[].
```

Comment:
I suggest to:
 - merge the "Ping Timeout" and "Ping Gather Job Timeout" sections under one title
 - start with "During Salt batch calls, the Salt master will perform a presence ping to target minions."
 - explain the presence ping purpose
 - continue with something like:

"As an ordinary Salt command, the presence ping is subject to the same kind of timeouts that affect all Salt commands, namely the main `timeout` and the `gather_job_timeout` - their meaning in detail is explained at `xref...`.

However, the default lengths of these timeouts are significantly shorter for presence pings, to allow quicker detection of unresponsive minions. They can also be configured individually, via the following parameters in `rhn.conf`..."

Reply:
We can also include, in the "presence ping purpose" section, that the execution of the batching calls will start when either all the minions replied before the presence ping timeout, or we actually reach the time out and we execute it with the minions that  reply. What happens first

Reply:
OK, I've combined the batching bit as well, and made them more or less parallel. Think I  got everything!